test/e2e/specs/editor/various/patterns.spec.js

```diff
@@ -74,28 +65,52 @@ test.describe( 'Unsynced pattern', () => {
 				name: newCategory,
 			} )
 			.click();
+
 		const pattern = page.getByLabel( 'My unsynced pattern' ).first();
 
 		const insertedPatternId = await pattern.evaluate(
 			( element ) => element.id
 		);
 
-		await pattern.click();
+		// Check that the block content is still the same. If the pattern was added as synced
+		// the content would be wrapped by a pattern block.
+		const expectedUnsyncedPattern = {
+			...before[ 0 ],
+			attributes: {
+				...before[ 0 ].attributes,
+				metadata: {
+					name: 'My unsynced pattern',
+					patternName: insertedPatternId,
+					// When a pattern is created for the first time the `categories` are missing.
+					// This is a known issue that needs to be fixed.
```

Comment:
As well as adding the `patternName`, we should add a `categories` apparently.

The way it works is a bit hard to understand - sometimes the category id is added, sometimes the category name.

I can't figure out what `categories` property does, so for now I've not added it. I'll make a follow-up issue to add it consistently.

Reply:
I have a very old repressed memory that categories here need to handle both the "default" categories (ones with a name) and those created by the user which leverage a taxonomy and end up with IDs.

A quick search dug up these that might help shine a light on the situation:
- https://github.com/WordPress/gutenberg/issues/53164
- https://github.com/WordPress/gutenberg/pull/54640

Reply:
Thanks for the links, that'll help for sure! I'll make sure to add them to the issue I create.